nri-prelude/tests/TaskSpec.hs

```diff
@@ -0,0 +1,63 @@
+module TaskSpec (tests) where
+
+import qualified Expect
+import qualified List
+import NriPrelude
+import qualified Platform.Internal as Internal
+import Prelude ((*>), pure)
+import qualified Process
+import qualified Result
+import qualified System.TimeIt
+import Test (Test, describe, test)
+import Test.Internal (Failure)
+import qualified Task
+import qualified Tuple
+
+tests :: Test
+tests =
+  describe "Task"
+    [ describe "parallel"
+      [ test "returns the right values and executes in parallel" <| \() -> do
+        let results = [1, 2, 3]
+        let tasks = List.map afterDelay results
+
+        (parallelExecutionTime, parallelResults) <-
+          Expect.succeeds <| timeIt <| Task.parallel tasks
+        (sequentialExecutionTime, _) <-
+          Expect.succeeds <| timeIt <| Task.sequence tasks
+
+        Expect.equal results parallelResults
+
+        parallelExecutionTime |>
+          Expect.lessThan sequentialExecutionTime
+      ]
+    , describe "concurrently"
+      [ test "returns the right values and executes concurrently" <| \() -> do
+        let results = (1, ("two", 3.0 :: Float))
+        let (taskA, (taskB, taskC)) = Tuple.mapBoth afterDelay (Tuple.mapBoth afterDelay afterDelay) results
+
+        (concurrentExecutionTime, concurrentResults) <-
+          Expect.succeeds <| timeIt <| Task.concurrently taskA <| Task.concurrently taskB taskC
+        (sequentialExecutionTime, _) <-
+          Expect.succeeds <| timeIt <| Task.andThen (\_ -> taskB) taskA
+
+        Expect.equal results concurrentResults
+
+        concurrentExecutionTime |>
+          Expect.lessThan sequentialExecutionTime
```

Comment:
Is this a bad idea (asking for a flake)?
Probably -- I guess its success depends on being compiled with `-threaded` and executed with `-N`.

I'm going to remove the comparisons of execution time.

Reply:
Well, yes, it depends on `-threaded` and `-N`, since the test suite is here we have control over those things. So I think I changed my mind and will put it back.

Still not sure why the parallel/concurrent cases were faster than the single cases 🤷 .

Reply:
Hmm, also depends on being run on a machine with more than one core available, I guess... probably a bad idea... open to input here.

Reply:
I'm going to remove it on the grounds that that's more of a benchmark than a test.

Reply:
> Is this a bad idea (asking for a flake)? ...

That was going to be my only input TBH. I think testing primitives is hard in that sense. We're currently not at our desired scope of investigation. I'm more than happy with the test for the equivalence between `parallel` and `concurrently`